packages/metro-resolver/src/__tests__/package-exports-test.js

```diff
@@ -225,14 +226,14 @@ describe('with package exports resolution enabled', () => {
       test('without expanding `sourceExts`', () => {
         expect(Resolver.resolve(context, 'test-pkg', null)).toEqual({
           type: 'sourceFile',
-          filePath: '/root/node_modules/test-pkg/index-main.js',
+          filePath: '/root/node_modules/test-pkg/index-exports.js',
```

Comment:
With the other comment and change, I think this is the right assertion instead. Note that, with no option to disable strict behaviour today, we can't assert the resolver throws an error in this case. But instead, a comment makes this sufficiently easy to understand, IMO.

```suggestion
          // [nonstrict] Falls back to index.js based on file resolution
          filePath: '/root/node_modules/test-pkg/index.js',
```

Reply:
Hmm regarding this, I think we should just add another test cases for testing the fallback, as right now with changes from this PR we directly test not expanding `sourceExts` & `platformExts` while falling back proves the functionality but indirectly, if that makes any sense, WDYT?

Reply:
It is indirect, but is the only observable behaviour we have right now 😅 We could also add a `logWarning` assertion in these to make even clearer — but I think the comments are good enough.